x/derivatives/keeper/positions.go

```diff
@@ -105,11 +124,21 @@ func (k Keeper) DeletePosition(ctx sdk.Context, address sdk.AccAddress, id strin
 }
 
 func (k Keeper) OpenPosition(ctx sdk.Context, msg *types.MsgOpenPosition) error {
-	sender := msg.Sender.AccAddress()
-	lastPositionId := k.GetLastPositionId(ctx)
+	// todo check sender amount for margin
 
-	positionKey := types.AddressPositionWithIdKeyPrefix(sender, lastPositionId)
-	positionId := string(positionKey)
+	lastPosition, err := k.GetLastPosition(ctx)
+	if err != nil {
+		panic("failed to get last position")
```

Comment:
maybe better to define Error case and return err, not panic?

Reply:
This case is usually not possible.
In such cases, Chain should be dropped.

Reply:
Sorry, I made a mistake.
I've fixed it